src/V3Fork.cpp

```diff
@@ -463,6 +466,7 @@ class ForkVisitor final : public VNVisitor {
     AstVar* m_capturedVarsp = nullptr;  // Local copies of captured variables
     std::set<AstVar*> m_forkLocalsp;  // Variables local to a given fork
     AstArg* m_capturedVarRefsp = nullptr;  // References to captured variables (as args)
+    size_t m_id = 0;  // Unique ID for a task
```

Comment:
A global counter is better than using the node address, but know that it tends to destroy ccache hit rates on re-builds, as adding a single new entry might change all variable names, which is why we use hashing in most other places.

Reply:
BTW I'm not objecting to merging this, just an FYI

Reply:
Though IDs are less preferred, when they are used please reset on every AstNodeModule.